handbook/engineering/distribution/observability/monitoring_architecture.md

```diff
@@ -0,0 +1,118 @@
+
+We use [Alertmanager](https://prometheus.io/docs/alerting/latest/alertmanager/) for:
+
+- Providing data about currently active Sourcegraph alerts.
+- Routing alerts to appropriate receivers and silencing them when desired, [configured using site configuration](#alert-notifications).
+
+Alertmanager is bundled in `sourcegraph/prometheus`, and notifications are configured for Sourcegraph alerts [using site configuration](https://docs.sourcegraph.com/admin/observability/alerting). This functionality is provided by the [prom-wrapper](#prom-wrapper).
+
+*Rationale for notifiers in site configuration*: Due to the limitations of [admin reverse-proxies](#admin-reverse-proxy), alerts cannot be configured without port-forwarding or custom ConfigMaps, something we [want to avoid](monitoring_pillars.md#long-term-vision).
+
+*Rationale for Alertmanager*: An approach for notifiers using Grafana was considered, but had some issues outlined in [#11832](https://github.com/sourcegraph/sourcegraph/pull/11832), so Alertmanager was selected as our notification provider.
+
+*Rationale for silencing in site configuration*: Similar to the [Grafana admin reverse-proxy](#admin-reverse-proxy), silencing using the Alertmanager UI would require port-forwarding, something we [want to avoid](monitoring_pillars.md#long-term-vision).
+
+#### prom-wrapper
+
+The [prom-wrapper](https://github.com/sourcegraph/sourcegraph/tree/master/docker-images/prometheus/cmd/prom-wrapper) is the entrypoint program for `sourcegraph/prometheus` and it:
+
+* Handles starting up Prometheus and Alertmanager
+* Applies updates to site configuration by [generating a diff and applying changes](https://sourcegraph.com/search?q=repo:%5Egithub.com/sourcegraph/sourcegraph%24+file:docker-images/prometheus+type:symbol+Change+OR+Diff&patternType=literal)
+  * Most notably, this includes [configuring notifiers and silences](#alert-notifications) for Sourcegraph alerts
+* Exposes [endpoints for configuration issues, alerts summary statuses, and reverse-proxies Prometheus and Alertmanager](https://sourcegraph.com/search?q=repo:%5Egithub.com/sourcegraph/sourcegraph%24+file:docker-images/prometheus+PathPrefix%28:%5Bpath%5D%29.Handler%28:%5Bhandler%5D%29&patternType=structural)
+
+*Rationale for an all-in-one Prometheus image with prom-wrapper*: This allows us to avoid adding a new separate service (which must be handled in all our deployment types), thus simplifying the deployment story for our monitoring stack, while also improving the alert debugging workflow (for example, simply port-forward `svc/prometheus` to get access to the entire alerting stack), with minimal disadvantages (for example, high-availability Prometheus and Alertmanager can still be configured via `PROMETHEUS_ADDITIONAL_FLAGS` and `ALERTMANAGER_ADDITIONAL_FLAGS`, and Alertmanager can be disabled via `DISABLE_ALERTMANAGER`).
+
+## Custom additions
+
+TODO: how we handle out-of-band metrics, alerts (things we don't ship to customers)
```

Comment:
@slimsag From what I understand, this is pretty ad-hoc "add more prometheus yaml" at the moment - is there anything I'm missing?

If there's no current process for this, I'd like to get an RFC started about a set of metric/alert labels/standards for out-of-band metrics and alerts that can integrate with the built-in alerting stack. This will somewhat depend on how https://github.com/sourcegraph/sourcegraph/issues/12010 comes out

Reply:
For example, oddity with blackbox alerts: https://sourcegraph.slack.com/archives/CJX299FGE/p1595254363107300

Reply:
update: started sketching out [RFC 208](https://docs.google.com/document/d/1ub1xsrKTdjA5n-4MAk4AZXtInmgJlMwJ0pI1opkq5mo/), but probably will keep this on hold given our new priorities